src/components/battles/animations/shared/Fainting.ts

```diff
@@ -45,19 +45,48 @@ export class Fainting<TGameStartr extends FullScreenPokemon> extends GeneralComp
                 const playerName = this.gameStarter.itemsHolder.getItem("name");
                 const partyIsWipedText: (string | string[])[][] = [[pokemon.nickname, " fainted!"]];
 
-                this.gameStarter.physics.killNormal(thing);
-                this.gameStarter.physics.killNormal(blank);
-                this.gameStarter.menuGrapher.createMenu("GeneralText");
-                if (this.gameStarter.battles.isPartyWiped()) {
-                    partyIsWipedText.push(
-                        [playerName, " is out of useable Pokemon!"],
-                        [playerName, " blacked out!"]);
-                }
-
-                this.gameStarter.menuGrapher.addMenuDialog("GeneralText", partyIsWipedText, onComplete);
-                this.gameStarter.menuGrapher.setActiveMenu("GeneralText");
+                if (teamName === "opponent") {
+                    this.processOpponentFainting(partyIsWipedText, onComplete, battleInfo, thing, blank);
+                } else {
+                    this.processPlayerFainting(partyIsWipedText, onComplete, battleInfo, thing, blank, playerName);
+            }
             });
 
         this.gameStarter.modAttacher.fireEvent(this.gameStarter.mods.eventNames.onFaint, pokemon, battleInfo.teams.player.actors);
     }
+     /**
+      * Helper function to start the process of gaining experience
+      */
+    private processOpponentFainting(
+        partyIsWipedText: (string | string[])[][],
+        onComplete: () => void, battleInfo: IBattleInfo,
+        thing: IThing, blank: IThing) {
+        this.gameStarter.menuGrapher.createMenu("GeneralText");
+        this.gameStarter.menuGrapher.addMenuDialog(
+            "GeneralText", partyIsWipedText,
+            () => this.gameStarter.experience.processBattleExperience(battleInfo, onComplete));
+        this.gameStarter.menuGrapher.setActiveMenu("GeneralText");
+        this.gameStarter.physics.killNormal(thing);
+        this.gameStarter.physics.killNormal(blank);
```

Comment:
You `killNormal` the `thing` and `blank` in both of these private functions. Remove code duplication by doing it once in `run`.